flink-runtime/src/test/java/org/apache/flink/runtime/security/token/ExceptionThrowingDelegationTokenProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.security.token;
+
+import org.apache.flink.configuration.Configuration;
+
+import org.apache.hadoop.security.Credentials;
+
+import java.util.Optional;
+
+/**
+ * An example implementation of {@link DelegationTokenProvider} which throws exception when enabled.
+ */
+public class ExceptionThrowingDelegationTokenProvider implements DelegationTokenProvider {
+
+    public static volatile boolean enabled = false;
```

Comment:
Not a big fan of this approach as it's fairly fragile and the tests that are using the class can not be parallelized. 🤔

It's probably ok-ish as long as this is used only by a single test class.

Reply:
I fully agree that it's ugly but I haven't had better idea which is not an overkill. If this would be used from multiple places then something different is needed for sure.